### ISSUE TITLE
OpenTracing: support pluggable strategy for span naming/tagging

### ISSUE THREAD

**Issue by sdwr98** [temporalio/sdk-java#502]
We use DataDog for APM and tracing, and they do support integration with OpenTracing.  However, they expect span naming and tagging to be in a particular format that does not align with the current Temporal OT implementation

Currently the Temporal OT implementation names the spans like `RunWorkflow_<WorkflowName>` and `RunActivity_<ActivityName>`, and adds tags for Workflow/Run ID.   However, DataDog expects a span name like `RunWorkflow` or `RunActivity` and a tag with the workflow/activity name. 

It would be great if the `OpenTracingOptions` provided the ability to completely override the strategy for naming/tagging spans.

**Comment by Spikhalskiy** [comment]
We discussed with Max during the implementation of the initial open tracing support that it's a good idea to support custom strategies and add an interface that gives full freedom on span names/tags creation. JaxRS/Jersey implementation has an alternative strategy similar to what described in this feature request: https://github.com/opentracing-contrib/java-jaxrs/blob/master/opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java#L25
We just decided to postpone this until somebody actually asks for alternative implementations. 
So, I support the idea of creating an `OperationNameAndTagsProvider` that provides span name and tags based on some context and adding more standard implementation. 

I have a strong opinion only that the strategy that we have already implemented (RunWorkflow_<WorkflowName> and RunActivity_<ActivityName>) should be provided and most likely should be considered default based on my evaluation of other opentracing/opentelemetry implementations. See JaxRS/Jersey: https://github.com/opentracing-contrib/java-jaxrs/blob/master/opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java#L46
